### PR TITLE
Add storefront sdk lib

### PR DESCRIPTION
**Pull request by dhmacs**
This PR adds a Typescript sdk to have typed queries for the Shopify Storefront GraphQL api (similar to what we do for Strapi GraphQL api)

Nothing changed in term of functionality.

### CODE REVIEW

File: frontend/lib/shopify-storefront-sdk/index.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+         }
+         throw new Error('Data not available in GraphQL response');
+      }
+      if (Array.isArray(result.errors)) {
+         console.error('GraphQL query failed with errors:');
+         result.errors.map((error: any) => {
+            const code = error.extensions?.code || 'UNKNOWN';
+            console.error(`\t[${code}]`, error.message);
+         });
+      }
```
Comment (masonmcelvain) [inline frontend/lib/shopify-storefront-sdk/index.ts:46]: Should we report this case to Sentry since we don't throw any new errors?

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻 Nice work! I have a question, but I'm happy merging this as-is.

File: frontend/lib/shopify-storefront-sdk/index.ts (hunk 1 above)
Comment (dhmacs) [reply frontend/lib/shopify-storefront-sdk/index.ts:46]: @masonmcelvain yes, I think so. I checked and we're not reporting errors for the strapi-sdk either. So it's probably worth to do it in a separate PR for both use cases. If you'd like feel free to open a PR 👍

**Comment by erinemay** [comment]
Adding this here from @masonmcelvain 
> For QA, I'd suggest running the new command and making sure there are no errors and that the correct file gets generated. You could even try changing the findProduct.graphql query and then running it again, and making sure that the output types are updated correctly. Besides that, I don't think there's much else to QA.

>pnpm codegen:download-shopify-storefront-schema it's in the package.json as a new script

**Comment by erinemay** [comment]
Is there any additional setup required to get the new command to run? Or a specific place to run it from? 

I'm getting 127

```Shell Scripting
erin13mm@DESKTOP-LSSD59U:~/react-commerce$ cd frontend && pnpm run codegen:download-shopify-storefront-schema

> @ifixit/commerce-frontend@1.0.0 codegen:download-shopify-storefront-schema /home/erin13mm/react-commerce/frontend
> node ./lib/shopify-storefront-sdk/generate-schema.js

ERROR: 127

```

And my server output after running it is:

```

[dev:frontend] [codegen] [10:30:02] Parse configuration [started]
[dev:frontend] [codegen] [10:30:02] Parse configuration [completed]
[dev:frontend] [codegen] [10:30:02] Generate outputs [started]
[dev:frontend] [codegen] [10:30:02] Generate lib/strapi-sdk/generated/sdk.ts [started]
[dev:frontend] [codegen] [10:30:02] Generate lib/shopify-storefront-sdk/generated/sdk.ts [started]
[dev:frontend] [codegen] [10:30:02] Load GraphQL schemas [started]
[dev:frontend] [codegen] [10:30:02] Load GraphQL schemas [started]
[dev:frontend] [codegen] [10:30:02] Load GraphQL schemas [failed]
[dev:frontend] [codegen] [10:30:02] → Failed to load schema
[dev:frontend] [codegen] [10:30:02] Generate lib/shopify-storefront-sdk/generated/sdk.ts [failed]
[dev:frontend] [codegen] [10:30:02] → Failed to load schema
[dev:backend ] [2022-08-24 10:30:02.255] http: POST /graphql (27 ms) 200
[dev:frontend] [codegen] [10:30:02] Load GraphQL schemas [completed]
[dev:frontend] [codegen] [10:30:02] Load GraphQL documents [started]
[dev:frontend] [codegen] [10:30:02] Load GraphQL documents [completed]
[dev:frontend] [codegen] [10:30:02] Generate [started]
[dev:frontend] [codegen] [10:30:02] Generate [completed]
[dev:frontend] [codegen] [10:30:02] Generate lib/strapi-sdk/generated/sdk.ts [completed]
[dev:frontend] [codegen] [10:30:02] Generate outputs [failed]
[dev:frontend] [codegen]   ℹ Watching for changes...

```

**Comment by dhmacs** [comment]
@erinemay did you install dependencies with
`pnpm install:all` ? Usually error 127 happens when it doesn't find the gq command

un_dev_block ❓

**Comment by erinemay** [comment]
Yeah, that was it. 

QA 🦖 
- Running the new command now successfully generated `lib/shopify-storefront-sdk/generated/sdk.ts`. Confirmed the file looks appropriate based on `frontend/lib/shopify-storefront-sdk/operations/findProduct.graphql`
- Store smoke test looks good

deploy_block ❔ on a question.
From the [development preview](https://react-commerce-git-add-storefront-sdk-ifixit.vercel.app/Parts), I would expect checkout to lead me to the ifixit-test shopify store (store.cominor). When I try to check out from the mini cart on this branch, I'm sent to store.ifixit.com which immediately redirects to https://www.ifixit.com/Store. Has there been a change in the development preview, or is this possibly a result of the changes here?

**Comment by masonmcelvain** [comment]
@erinemay I see this behavior on the dev preview for `main`, so I don't think it's related to this feature. Good catch though.

un_deploy_block ✌🏻

**Comment by dhmacs** [comment]
> From the [development preview](https://react-commerce-git-add-storefront-sdk-ifixit.vercel.app/Parts), I would expect checkout to lead me to the ifixit-test shopify store (store.cominor). When I try to check out from the mini cart on this branch, I'm sent to store.ifixit.com which immediately redirects to https://www.ifixit.com/Store. Has there been a change in the development preview, or is this possibly a result of the changes here?

@erinemay Basically it's due to the fact that the Strapi seed plugin now is copying data over from `strapi.ifixit.com` and that includes the shopify store settings. Right now the solution is to manually update those values. 
Long term solution is to tweak the seed plugin to account for these edge cases where we don't want to copy production settings.

**Pull request merged**